src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageItemDeprecationLabel.xaml.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI
+{
+    /// <summary>
+    /// Interaction logic for PackageItemDeprecationLabel.xaml
+    ///
+    /// DataContext is <see cref="PackageDeprecationMetadataContextInfo"/>
+    /// 
+    /// Similar to <see cref="PackageManagerProvidersLabel"/>
+    /// </summary>
+    public partial class PackageItemDeprecationLabel : UserControl
+    {
+        public PackageItemDeprecationLabel()
+        {
+            InitializeComponent();
+        }
+
+        private string _formatStringSingle;
+        public string FormatStringSingle
+        {
+            get => _formatStringSingle;
+
+            set
+            {
+                if (_formatStringSingle != value)
+                {
+                    _formatStringSingle = value;
+                    FillTextBlock();
+                }
+            }
+        }
+
+        private string _formatStringAlternative;
+        public string FormatStringAlternative
+        {
+            get => _formatStringAlternative;
+
+            set
+            {
+                if (_formatStringAlternative != value)
+                {
+                    _formatStringAlternative = value;
+                    FillTextBlock();
+                }
+            }
+        }
+
+        private void UserControl_DataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            FillTextBlock();
+        }
+
+        private void FillTextBlock()
```

Comment:
It would be great if we could generalize this as a reusable utility method living outside of this class, that decorates a portion of a given string with a hyperlink, and put the method under unit tests.

Reply:
Moved UI logic to XAML and added a new WPF Value converter to handle control state.

Reply:
Added converter unit tests.

Reply:
Added tests to new converters.